lib/libdl/dlfcn.c

```diff
@@ -0,0 +1,121 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2019 Linaro limited
+ */
+
+#include <ctype.h>
+#include <dlfcn.h>
+#include <pta_system.h>
+#include <stdlib.h>
+#include <string.h>
+#include <tee_api.h>
+#include <tee_internal_api_extensions.h>
+
+static TEE_TASessionHandle sess = TEE_HANDLE_NULL;
```

Comment:
It seems there's no way to closing this handle once it has been opened.

Reply:
Correct, and we have the same pattern twice in `libutee` already (`invoke_socket_pta()` and `invoke_system_pta()`). Perhaps it's not worth caching the session? I could use the ftrace framework to get an idea of the overhead, now that it records execution time.